sbids/tasks/lotkavolterra.py

```diff
@@ -0,0 +1,178 @@
+import jax
+from jax import lax
+import jax.numpy as jnp
+from jax.experimental.ode import odeint
+from tensorflow_probability.substrates import jax as tfp
+tfd = tfp.distributions
+tfb = tfp.bijectors
+
+
+class Lotka_Volterra:
+
+    def __init__(self,
+                num_seed:int=None):
+        self.num_seed = num_seed
+
+        def equation(y,t, theta):
+            """
+            Evaluate the Lotka-Volterra time derivative of the solution `y` at time
+            `t` as `func(y, t, *args)` to feed odeint.
+            Parameters
+            ----------
+            y: float
+            The two populations.
+            t : float
+            The time.
+            theta: float
+            Model parameters.
+            Returns
+            -------
+            Lotka-Volterra equations.
+            """
+            X = y[0]
+            Y = y[1]
+
+            alpha, beta, gamma, delta = (
+                theta[..., 0],
+                theta[..., 1],
+                theta[..., 2],
+                theta[..., 3],)
+
+            dX_dt = alpha * X - beta * X * Y
+            dY_dt = -gamma * Y + delta * X * Y
+            return jnp.stack([dX_dt, dY_dt],axis=0)
+        self.equation = equation
+    
+        @jax.jit
+        def solve_equation(init,theta):
+          
+            ts = jnp.arange(0.,10.,0.1)
+            z = odeint(equation, init, ts, theta, rtol=1e-9, atol=1e-9)
+            z = z.T[:,::21]
+
+            return z
+        self.solve_equation = solve_equation
+
+
+    def get_truth(self):
```

Comment:
this function should be function of an rng key, that gets splitted.

Reply:
and actually, I think there should be no difference between simulation and truth. It's just a draw from from the probabilistic model.